lms/static/js/spec/student_account/account_settings_fields_spec.js

```diff
@@ -0,0 +1,320 @@
+                        helpMessage: 'This is your login username.'
+                    })
+
+                    var view = new fieldView(fieldData).render();
+
+                    view.showHelpMessage();
+                    expectMessageContains(view, view.helpMessage);
+                    view.showSuccessMessage();
+                    expectMessageContains(view, view.successMessagePrefix);
+                    jasmine.Clock.tick(5000);
+                    // Message gets reset
+                    expectMessageContains(view, view.helpMessage);
+
+                    view.showSuccessMessage();
+                    expectMessageContains(view, view.successMessagePrefix);
+                    // But if we change the message, it should not get reset.
+                    view.message("Do not reset this!");
+                    jasmine.Clock.tick(5000);
+                    expectMessageContains(view, "Do not reset this!");
+                }
+            });
+
+            it("sends a PATCH request when saveAttributes is called", function() {
+
+                requests = AjaxHelpers.requests(this);
+
+                var fieldView = AccountSettingsFieldViews.FieldView;
+                var fieldData = createFieldData(fieldView, {
+                    title: 'Preferred Language',
+                    valueAttribute: 'language',
+                    helpMessage: 'Your preferred language.'
+                })
+
+                var view = new fieldView(fieldData);
+                view.saveAttributes({'language': 'ur'}, {
+                    'headers': {'Priority': 'Urgent'}
+                });
+
+                var request = requests[0];
+                expect(request.method).toBe('PATCH');
+                expect(request.requestHeaders['Content-Type']).toBe('application/merge-patch+json;charset=utf-8');
+                expect(request.requestHeaders['Priority']).toBe('Urgent');
+                expect(request.requestBody).toBe('{"language":"ur"}');
+            });
+
+            it("correctly renders ReadonlyFieldView", function() {
+                var fieldData = createFieldData(AccountSettingsFieldViews.ReadonlyFieldView, {
+                    title: 'Username',
+                    valueAttribute: 'username',
+                    helpMessage: 'This is your login username'
+                });
+                var view = new AccountSettingsFieldViews.ReadonlyFieldView(fieldData).render();
+
+                expectTitleAndMessageToBe(view, fieldData.title, fieldData.helpMessage);
+                expect(view.$('.account-settings-field-value').text().trim()).toBe(MOCK_USERNAME);
+            });
+
+            it("correctly updates ReadonlyFieldView on model update", function() {
+                var fieldData = createFieldData(AccountSettingsFieldViews.ReadonlyFieldView, {
+                    title: 'Username',
+                    valueAttribute: 'username',
+                    helpMessage: 'This is your login username'
```

Comment:
Is this a tooltip? If so, I'd change the text to "The username that you use to sign in to edX"